harness/determined/deploy/gke/cli.py

```diff
@@ -0,0 +1,551 @@
+                machine_type, zone
+            ),
+        )
+
+
+def validate_accelerator_zone(args: argparse.Namespace, zone: str) -> None:
+    try:
+        subprocess.check_call(
+            [
+                "gcloud",
+                "compute",
+                "accelerator-types",
+                "describe",
+                args.gpu_type,
+                "--zone",
+                zone,
+            ],
+            stdout=subprocess.DEVNULL,
+        )
+    except:
+        raise ValueError(
+            "The specified accelerator type {} was not found in the specified zone {}.".format(
+                args.gpu_type, zone
+            ),
+        )
+
+
+def validate_args(args: argparse.Namespace) -> None:
+    validate_location(args.zone, isZone=True)
+    validate_accelerator_type(args.gpu_type)
+    if args.master_machine_type != "n1-standard-16":
+        validate_machine_type(args.master_machine_type, args.zone)
+    if args.agent_machine_type != "n1-standard-32":
+        validate_machine_type(args.agent_machine_type, args.zone)
+    validate_accelerator_zone(args, args.zone)
+    if args.multi_node_pool:
+        if args.cpu_node_pool_name is None:
+            raise ValueError(
+                "--cpu-node-pool-name must be specified if using multiple node pools "
+                + "(--multi-node-pool flag is set).",
+            )
+    if args.gpu_coscheduler and args.preemption:
+        raise ValueError(
+            "--gpu-coscheduler and --preemptive-scheduler are mutually exclusive and cannot both be"
+            " specified",
+        )
+    if args.cpu_coscheduler and not args.gpu_coscheduler:
+        raise ValueError(
+            "To enable coscheduling on the CPU Node Pool, coscheduling must be enabled on the GPU"
+            " Node Pool. --cpu-coscheduler cannot be specified if --gpu-coscheduler isn't as well."
+        )
+    if args.cpu_coscheduler and not args.multi_node_pool:
+        raise ValueError(
+            "To enable coscheduling on the CPU Node Pool, multiple node pools must be used."
+            + " --cpu-coscheduler cannot be specified if --multi-node-pool isn't as well."
+        )
+
+
+def create_cluster(args: argparse.Namespace) -> None:
+    region = "-".join(args.zone.split("-")[:-1])
+    cmd = [
+        "gcloud",
+        "container",
+        "clusters",
+        "create",
+        args.cluster_name,
+        "--region",
+        region,
+        "--node-locations",
+        args.zone,
+        "--num-nodes=1",
+        "--machine-type={}".format(args.master_machine_type),
+    ]
+    subprocess.check_call(cmd, stdout=subprocess.DEVNULL)
+    create_nodepools(region, args)
+    if not args.no_make_bucket:
+        cmd = ["gsutil", "mb", "gs://{}".format(args.gcs_bucket)]
+        subprocess.check_call(cmd, stdout=subprocess.DEVNULL)
+
+
+def create_nodepools(region: str, args: argparse.Namespace) -> None:
+    cmd = [
+        "gcloud",
```

Comment:
For the purpose of knowledge-sharing: this mostly comes up with A100s at the moment, but sometimes this will appear to fail because of capacity, even with 0 min nodes, and if so at a later point you could try to actually provision nodes and possibly be successful (or vice versa). I'm torn on whether or not the right UX is to actually treat that as a fatal error or just raise a warning about capacity.